### PR TITLE
Fix issues formatting query strings.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?

This pull request follows up on the quick fix from #299 with an improved `querify` helper function that correctly omits null values from the query string. This addresses the same bug described in that PR in a more generalized & clearer way.

### How should this be reviewed?

I refactored our resolvers to use this function (instead of string interpolation or directly calling `stringify`) in 9b6acbf, then added a test demonstrating the old issue we were facing in 659aa97. I then "re-fixed" the bug in ab9a1ab.

I also fixed an unrelated issue where GraphQL errors would not be clearly reported in test failures in 6eb3aba.

### Any background context you want to provide?

At some point, I believe we started getting `null` for unused variables instead of `undefined` & so `qs`'s default behavior no longer stripped them from the query string. I can't find this change in `graphql`, `apollo-server`, or `apollo-client`'s changelogs, but I imagine it must have been one of them...

### Relevant tickets

References [Pivotal #175669154](https://www.pivotaltracker.com/story/show/175669154).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: src/repositories/helpers.js [hunk 1]
```diff
@@ -141,3 +142,12 @@ export const getOptional = (schema, type) => {
...
 export const withoutNil = data => omitBy(data, isNil);
+
+/**
+ * Create a query string from the given object.
+ *
+ * @param {Object} values
+ * @return {String}
+ */
+export const querify = queryValues =>
+  stringify(queryValues, { skipNulls: true, addQueryPrefix: true });
```
Comment (mendelB) [inline src/repositories/helpers.js:153]: `addQueryPrefix ` will this auto prepend the `?`?

File: src/repositories/rogue.js [hunk 2]
```diff
@@ -379,7 +407,7 @@ export const toggleReaction = async (postId, context) => {
 export const updatePostQuantity = async (postId, quantity, context) => {
   const response = await fetch(`${ROGUE_URL}/api/v3/posts/${postId}`, {
     method: 'PATCH',
-    body: JSON.stringify({ quantity }),
+    body: JSON.querify({ quantity }),
```
Comment (mendelB) [inline src/repositories/rogue.js:410]: How do we end up with this `querify` method on the `JSON` object? (I don't think that's standard is it?)

File: src/repositories/rogue.js [hunk 3]
```diff
@@ -594,11 +622,19 @@ export const getSignupsById = async (ids, options) => {
     ids,
   });
 
-  const idQuery = ids.join(',');
+  const queryString = {
+    filter: {
+      ids: ids.join(','),
```
Comment (mendelB) [inline src/repositories/rogue.js:627]: I think this needs to be `id` singular?
```suggestion
      id: ids.join(','),
```

**Review by mendelB — APPROVED** [review]
👏 Was fun to journey through those commits!

File: src/repositories/rogue.js (hunk 2 above)
Comment (DFurnes) [reply src/repositories/rogue.js:410]: We ended up with it by a _typo_… yikes! 🙈 That makes me want to fill in more integration tests.

File: src/repositories/rogue.js (hunk 3 above)
Comment (DFurnes) [reply src/repositories/rogue.js]: Yes, thank you!

File: src/repositories/helpers.js (hunk 1 above)
Comment (DFurnes) [reply src/repositories/helpers.js:153]: Yes! I just stumbled upon it when looking for `skipNulls`, but it should keep us from having any weird `/path?` URLs if we ever had an empty query string object.

File: src/repositories/rogue.js (hunk 2 above)
Comment (DFurnes) [reply src/repositories/rogue.js]: Fixed in 7be6c72. I think I'd been too aggressive with a find-replace on that.

**Comment by DFurnes** [comment]
Noticed another little goof in e472960 where I'd missed processing an object into a query string.

**Review by mendelB — APPROVED** [review]
✨

**Pull request merged**